test/lib/get-group-tangle.test.js

```diff
@@ -286,3 +287,36 @@ test('get-group-tangle with branch', t => {
     })
   }
 })
+
+test('members tangle', async t => {
+  const alice = Server()
+  const bob = Server()
+
+  const { groupId, root } = await p(alice.tribes.create)({})
+  const bobInvite = await p(alice.tribes.invite)(groupId, [bob.id], {})
+
+  const keystore = { group: { get: () => ({ root }) } }
+
+  const getGroupTangle = p(GetGroupTangle(alice, keystore))
+
+  const firstGroup = await getGroupTangle(groupId, 'group')
+  const firstMembers = await getGroupTangle(groupId, 'members')
+
+  t.deepEqual(firstGroup, { root, previous: [bobInvite.key] }, 'group tangle generated after add msg is correct')
+  t.deepEqual(firstMembers, { root, previous: [bobInvite.key] }, 'members tangle generated after add msg is correct')
```

Comment:
Weird that this is failing for both 'group' and 'members' tangles. I wonder what the third message they're pointing to is

```
members tangle
    ✖ 144) group tangle generated after add msg is correct
      @@ -1,5 +1,6 @@
       Object {
         "previous": Array [
      +    "%ShStVPrZ40tkb2i8wfDQUcG8caYXkn+yFffqKcN+NQc=.sha256",
           "%kQ16pXaNlyqVxBKt8xY9yN/GwCFJeSDe76wcsF2OVFA=.sha256",
         ],
         "root": "%7zX+ZdXus07//up94z1UDcAVaWddCuxVOzIixCT0bGU=.sha256",
      At: Test.<anonymous> (/test/lib/get-group-tangle.test.js:305:5)
      
    ✖ 145) members tangle generated after add msg is correct
      @@ -1,5 +1,6 @@
       Object {
         "previous": Array [
      +    "%ShStVPrZ40tkb2i8wfDQUcG8caYXkn+yFffqKcN+NQc=.sha256",
           "%kQ16pXaNlyqVxBKt8xY9yN/GwCFJeSDe76wcsF2OVFA=.sha256",
         ],
         "root": "%7zX+ZdXus07//up94z1UDcAVaWddCuxVOzIixCT0bGU=.sha256",
      At: Test.<anonymous> (/test/lib/get-group-tangle.test.js:306:5)
      
    ✔ exclude message gets tangles
    ✔ group tangle generated after exclude msg is correct
    ✔ members tangle generated after exclude msg is correct
```

Reply:
```diff
diff --git a/test/lib/get-group-tangle.test.js b/test/lib/get-group-tangle.test.js
index 5c30081..7fbf9d7 100644
--- a/test/lib/get-group-tangle.test.js
+++ b/test/lib/get-group-tangle.test.js
@@ -302,6 +302,12 @@ test('members tangle', async t => {
   const firstGroup = await getGroupTangle(groupId, 'group')
   const firstMembers = await getGroupTangle(groupId, 'members')
 
+  firstGroup.previous.forEach(prev=> {
+    alice.get({id:prev, private:true}, (err, msg)=> {
+      console.log('prev msg:', JSON.stringify(msg.content, null, 2))
+    })
+  })
+
   t.deepEqual(firstGroup, { root, previous: [bobInvite.key] }, 'group tangle generated after add msg is correct')
   t.deepEqual(firstMembers, { root, previous: [bobInvite.key] }, 'members tangle generated after add msg is correct')
 ```

gives these 2 from the same author:

```
prev msg: {
  "type": "group/add-member",
  "version": "v1",
  "groupKey": "L8P3E7uK8SaBB/9bbOqoEhw/kgryNQ7noDXuPnNtN4k=",
  "root": "%dMn/XE7f5xaFWv8KwPC7GKyVX7t96ha0vpU7AWEhfKI=.sha256",
  "tangles": {
    "members": {
      "root": "%dMn/XE7f5xaFWv8KwPC7GKyVX7t96ha0vpU7AWEhfKI=.sha256",
      "previous": [
        "%dMn/XE7f5xaFWv8KwPC7GKyVX7t96ha0vpU7AWEhfKI=.sha256"
      ]
    },
    "group": {
      "root": "%dMn/XE7f5xaFWv8KwPC7GKyVX7t96ha0vpU7AWEhfKI=.sha256",
      "previous": [
        "%dMn/XE7f5xaFWv8KwPC7GKyVX7t96ha0vpU7AWEhfKI=.sha256"
      ]
    }
  },
  "recps": [
    "%7C2lSl+1RaDYA/QFRrilKhc8ISSdgxcvXDcwRKPz5Qs=.cloaked",
    "@4ZYZR2fq6BaxsHflj7s1xGMswdocwRqsBQKbua/DNiI=.ed25519"
  ]
}
prev msg: {
  "type": "group/add-member",
  "version": "v1",
  "groupKey": "L8P3E7uK8SaBB/9bbOqoEhw/kgryNQ7noDXuPnNtN4k=",
  "root": "%dMn/XE7f5xaFWv8KwPC7GKyVX7t96ha0vpU7AWEhfKI=.sha256",
  "tangles": {
    "members": {
      "root": "%dMn/XE7f5xaFWv8KwPC7GKyVX7t96ha0vpU7AWEhfKI=.sha256",
      "previous": [
        "%dMn/XE7f5xaFWv8KwPC7GKyVX7t96ha0vpU7AWEhfKI=.sha256"
      ]
    },
    "group": {
      "root": "%dMn/XE7f5xaFWv8KwPC7GKyVX7t96ha0vpU7AWEhfKI=.sha256",
      "previous": [
        "%dMn/XE7f5xaFWv8KwPC7GKyVX7t96ha0vpU7AWEhfKI=.sha256"
      ]
    }
  },
  "recps": [
    "%7C2lSl+1RaDYA/QFRrilKhc8ISSdgxcvXDcwRKPz5Qs=.cloaked",
    "@cxwAVjzGhVnbWiDts4yALjzlv8p7F7Jpeo35HbsaJYw=.ed25519"
  ]
}
```

Reply:
it looks like the first one is alice adding herself, the second one is alice adding bob. but the add of bob for some reason only points to the root in the previous array. hmm this reminds me that if the tangle adding fails it just posts anyway, which would post with `previous: [root]`... :thinking:

Reply:
hmm no the publish hook doesn't seem to err early

Reply:
so this is really weird but i added a random timeout and now these tests are fixed *and* i found old tests that were passing incorrectly but are now fixed af11d47696983ae2d7268ce77b23a3fc97d2e059

Reply:
these 2 flake sometimes though :confused: